apache/superset
# fix(dashboard-export): Fixes  datasetId is not replaced with datasetUuid in Dashboard export in 4.1.x

### SUMMARY


Fixes #30424 by duplicating the code that rewrites the `native_filter_configuration` to use `datasetUuid` instead of `datasetId` into the `_file_content` method of `ExportDashboardsCommand`. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


Before the `targets` attribute of `native_filter_configuration` items includes `datasetId`:

```
    targets:
    - datasetId: 17
      column:
        name: column_name
```

After it just has `datasetUuid`:


```
    targets:
    - column:
        name: column_name
      datasetUuid: 7ac464dc-315a-4c9e-b9ec-048d7bbcf382```
```

### TESTING INSTRUCTIONS


(1) Create a dashboard with any chart
(2) Add a single value filter for any column
(3) Export the dashboard
(4) Verify that `datasetUuid` is used to refer to the dataset and not `datasetId`

### ADDITIONAL INFORMATION
